tests/gtfs/test_unmatched_id_warnings.py

```diff
@@ -0,0 +1,152 @@
+"""Tests for inserting invalid data into various GTFS tables."""
+import pytest
+import pandas as pd
+import numpy as np
+
+from transport_performance.gtfs.validation import GtfsInstance
+
+
+@pytest.fixture(scope="function")
+def gtfs_fixture():
+    """Test GtfsInstance() fixture."""
+    gtfs = GtfsInstance()
+    return gtfs
+
+
+@pytest.mark.runinteg
+class TestUmatchedIDWarnings(object):
+    """Tests for unmatch ID warnings in GTFS data."""
+
+    def test_unmatched_id_warnings_calendar(self, gtfs_fixture):
+        """Tests for unmatched IDs in the calendar table."""
+        # insert invalid data
+        gtfs_fixture.feed.calendar = pd.concat(
+            [
+                gtfs_fixture.feed.calendar,
+                pd.DataFrame(
+                    {
+                        "service_id": [101],
```

Comment:
These tests are sensitive to data format. Inserting service_id 101 rather than "101" is what error message "Invalid service_id; maybe has extra space characters" is complaining about.

Also, service_id "101" pre-exists in both calendar and trips. Switching the inserted service_id to "1666" (does not pre-exist) and running `feed.is_valid()`, no error is raised. 

Unmatched service_ids are tolerated in calendar.txt but not in trips.txt

Reply:
Point to note, if instead of adding an additional id to calendar, you remove an id in trips, no error is raised.

```
>>> feed.feed.trips = feed.feed.trips.drop(feed.feed.trips[feed.feed.trips["service_id"] == "195"].index)

>>> feed.is_valid()
      type                                            message       table                                               rows
1    error  Invalid route_type; maybe has extra space char...      routes               [1, 3, 4, 5, 6, 7, 8, 9, 12, 13, 14]
6    error                                  Undefined trip_id  stop_times  [80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 9...
0  warning                     Unrecognized column agency_noc      agency                                                 []
2  warning  Repeated pair (route_short_name, route_long_name)      routes                                           [11, 81]
3  warning                                 Route has no trips      routes                                                [0]
5  warning            Unrecognized column stop_direction_name  stop_times                                                 []
4  warning                  Unrecognized column platform_code       stops                                                 []
7  warning            Unrecognized column trip_direction_name       trips                                                 []
8  warning           Unrecognized column vehicle_journey_code       trips                                                 []


```